systems/trajectory_optimization/dircon_opt_constraints.cc

```diff
@@ -245,14 +250,47 @@ DirconKinematicConstraint<T>::DirconKinematicConstraint(
       type_{type}, is_constraint_relative_{is_constraint_relative},
       n_relative_{static_cast<int>(std::count(is_constraint_relative.begin(),
       is_constraint_relative.end(), true))} {
-  relative_map_ = MatrixXd::Zero(num_kinematic_constraints_, n_relative_);
-  int j = 0;
-  for (int i=0; i < num_kinematic_constraints_; i++) {
-    if (is_constraint_relative_[i]) {
-      relative_map_(i, j) = 1;
-      j++;
+  // ***Set sparsity pattern***
+  std::vector<std::pair<int, int>> sparsity;
+  // Acceleration constraints are dense in decision variables
+  for (int i = 0; i < num_kinematic_constraints_; i++) {
+    for (int j = 0; j < this->num_vars(); j++) {
```

Comment:
Maybe `for (int j = 0; j < num_states_ + num_kinematic_constraints_; j++)` is slightly faster?